src/view.c

```diff
@@ -267,6 +267,34 @@ view_minimize(struct view *view, bool minimized)
 	}
 }
 
+static void
+minimize_sub_views(struct view *view, bool minimized)
+{
+	struct view **child;
+	struct wl_array children = { 0 };
+
+	view_get_children(view, &children);
+	wl_array_for_each(child, &children) {
+		_minimize(*child, minimized);
+		minimize_sub_views(*child, minimized);
+	}
+	wl_array_release(&children);
+}
+
+/*
+ * Minimize the whole view-hierarchy from top to bottom regardless of which one
+ * in the hierarchy requested the minimize. For example, if an 'About' or
+ * 'Open File' dialog is minimized, its toplevel is minimized also. And vice
+ * versa.
+ */
+void
+view_minimize(struct view *view, bool minimized)
+{
+	struct view *root = view_get_root(view);
+	_minimize(root, minimized);
+	minimize_sub_views(root, minimized);
```

Comment:
`view_get_root()` seems to return `NULL` if `view->impl->get_root` is not set.
It would then call `_minimize()` which asserts `view`. Maybe we should return `view` instead of `NULL` in that case?

Reply:
Good catch. Done.